src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -535,26 +604,56 @@ private void InheritNullableStateOfFieldOrProperty(int targetContainerSlot, int
             }
         }
 
-        private void InheritNullableStateOfAnonymousTypeInstance(TypeSymbol targetType, int targetSlot, int valueSlot, bool isByRefTarget)
+        private void InheritNullableStateOfTrackableFieldOrPropertyType(int targetContainerSlot, int valueContainerSlot, Symbol fieldOrProperty, bool isByRefTarget)
+        {
+            int valueSlot = VariableSlot(fieldOrProperty, valueContainerSlot);
```

Comment:
>valueContainerSlot [](start = 58, length = 18)

Is it safe to call VariableSlot with ```valueContainerSlot``` <= 0 ? Is it going to return -1?

Reply:
It should handle that case but I've added a check to the caller to avoid that.